airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/ConfigInjector.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.config.persistence;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.micronaut.context.annotation.Requires;
+import jakarta.inject.Singleton;
+import java.io.IOException;
+import java.util.UUID;
+
+@Singleton
+@Requires(bean = ConfigRepository.class)
```

Comment:
As discussed, this will only create this singleton when the `ConfigRepository` is present as a singleton.  Because of this, you may want to inject `Optional<ConfigInjector>` everywhere it is used to avoid a possible null pointer.  You can simply modify the constructor parameter to take `Optional<ConfigInjector>` and Micronaut will do the rest.

Reply:
It is unavailable in the data-plane version of airbyte-worker.  In that case, we don't want to fail startup because we cannot create the ConfigInjector.

Reply:
Discussed this in a call, for now the current approach makes sense as cloud acceptance tests will catch the problem if someone tries to use this bean from the date plane.

Once the platform team tackles the project of removing all direct data base access from workers, an API will have to be introduced for it.